pilot/resource/titan.py

```diff
@@ -6,11 +6,65 @@
 #
 # Authors:
 # - Paul Nilsson, paul.nilsson@cern.ch, 2018
+# - Danila Oleynik danila.oleynik@cern.ch, 2018
 
 import logging
+import os
+import sys
+import shutil
+
+from pilot.util.config import config
+from pilot.util.filehandling import read_json, write_json, remove
+from pilot.common.exception import FileHandlingFailure
+from jobdescription import JobDescription
+
 logger = logging.getLogger(__name__)
 
 
+def get_rank():
+    rank = None
+    try:
+        from mpi4py import MPI
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+        max_rank = comm.Get_size()
+        logger.info("Got rank {0} from {1}".format(rank, max_rank))
+    except ImportError:
+        logger.info("No mpi4py found.")
+    return rank
+
+
+def get_job(harvesterpath):
+    """
+    Return job description in dictonary and MPI rank (if aplicable)
+    :return: job - dictonary with job description, rank
+    """
+    rank = 0
+    job = None
+    logger.info("Going to read job defenition from file")
+
+    pandaids_list_filename = os.path.join(harvesterpath, config.Harvester.jobs_list_file)
+    if not os.path.isfile(pandaids_list_filename):
+        logger.info("File with PanDA IDs is missed. Nothing to execute.")
+        return job, rank
+    harvesterpath = os.path.abspath(harvesterpath)
+    rank = get_rank()
+    pandaids = read_json(pandaids_list_filename)
+    logger.info('Got {0} job ids'.format(len(pandaids)))
+    pandaid = pandaids[rank]
+    job_workdir = os.path.join(harvesterpath, str(pandaid))
+
+    logger.info('Rank: {2} with job {0} will have work directory {1}'.format(pandaid, job_workdir, rank))
+
+    job_def_filename = os.path.join(job_workdir, config.Harvester.pandajob_file)
+    jobs_dict = read_json(job_def_filename)
+    job_dict = jobs_dict[str(pandaid)]
+    job = JobDescription()
```

Comment:
Danila, you rely on old `JobDescription` heavy implementation. I'd suggest to use `JobData` from the `infosys` component, 

https://github.com/PanDAWMS/pilot2/blob/next/pilot/info/jobdata.py
which is lightweight, with clean data mapping and definition, and support data validations, and attributes translation without any camels and snakes:)

Reply:
Either we replace the JobDescription with JobData which might take some time or I approve the pull request now so we get this in and fix the JobDescription later (but asap). Danila, can you make a quick estimate how long time it will take to fix?
About the flake8 issue with parse_jobreport_data(), a quick fix might be to just move out the DictQuery class and put it on the top of the module. Try that and then I can refactor the function properly (right away).

Reply:
I will try to migrate to JodData before end of next Monday. For 'parse_jobreport_data()'  i had add nested loops in one place - as dirty quick fix. I will refactor this part today, hope it will help.

Reply:
@anisyonk ,  I will be happy migrate from the "zoo"  to common solution :-)